libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4465,6 +4468,55 @@ void GCS_MAVLINK::send_set_position_target_global_int(uint8_t target_system, uin
             0,0);   // yaw, yaw_rate
 }
 
+void GCS_MAVLINK::send_smart_battery_info(const uint8_t instance) const
+{
+    const AP_BattMonitor &battery = AP::battery();
+
+	uint16_t cycles;
+	bool got_cycle_count = battery.get_cycle_count(instance,cycles);
+
+	int32_t capacity_full = battery.pack_capacity_mah(instance);
+
+    if (capacity_full == 0) {
+        capacity_full = -1;
+    }
+
+    char serial_number[16]{};
+    battery.get_serial_number(instance, serial_number, ARRAY_SIZE(serial_number)); 
+
+    char product_name[50]{}; 
```

Comment:
constants could be eliminated by using mavlink defines.
```suggestion
    char product_name[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_PRODUCT_NAME_LEN]{}; 
```

Reply:
Cool! Didn't know about those ones.